src/listener_builder.rs

```diff
@@ -170,35 +170,35 @@ macro_rules! make_tunnel_builder {
             /// [Using Labels]: https://ngrok.com/docs/guides/using-labels-within-ngrok/
             #[napi]
             pub fn label(&mut self, label: String, value: String) -> &Self {
-                let mut builder = self.tunnel_builder.lock();
+                let mut builder = self.listener_builder.lock();
                 builder.label(label, value);
                 self
             }
         }
     };
 }
 
-make_tunnel_builder! {
-    /// An ngrok tunnel backing an HTTP endpoint.
+make_listener_builder! {
+    /// An ngrok listener backing an HTTP endpoint.
     ///
-    /// @group Tunnel Builders
-    NgrokHttpTunnelBuilder, HttpTunnelBuilder, NgrokHttpTunnel, common
+    /// @group Listener Builders
+    HttpListenerBuilder, HttpTunnelBuilder, HttpListener, common
 }
-make_tunnel_builder! {
-    /// An ngrok tunnel backing a TCP endpoint.
+make_listener_builder! {
+    /// An ngrok listener backing a TCP endpoint.
     ///
-    /// @group Tunnel Builders
-    NgrokTcpTunnelBuilder, TcpTunnelBuilder, NgrokTcpTunnel, common
+    /// @group Listener Builders
+    TcpListenerBuilder, TcpTunnelBuilder, TcpListener, common
 }
-make_tunnel_builder! {
-    /// An ngrok tunnel backing a TLS endpoint.
+make_listener_builder! {
+    /// An ngrok listener backing a TLS endpoint.
     ///
-    /// @group Tunnel Builders
-    NgrokTlsTunnelBuilder, TlsTunnelBuilder, NgrokTlsTunnel, common
+    /// @group Listener Builders
+    TlsListenerBuilder, TlsTunnelBuilder, TlsListener, common
 }
-make_tunnel_builder! {
-    /// A labeled ngrok tunnel.
+make_listener_builder! {
+    /// A labeled ngrok listener.
     ///
-    /// @group Tunnel Builders
-    NgrokLabeledTunnelBuilder, LabeledTunnelBuilder, NgrokLabeledTunnel, label
+    /// @group Listener Builders
+    LabeledListenerBuilder, LabeledTunnelBuilder, LabeledListener, label
```

Comment:
why `LabeledTunnelBuilder` and above?

Reply:
or this is actually coming from `ngrok-rust`?

Reply:
Yeah, this is the reference to the underlying ngrok-rust builder to use